base/set.jl

```diff
@@ -427,6 +427,36 @@ allunique(::Union{AbstractSet,AbstractDict}) = true
 
 allunique(r::AbstractRange) = !iszero(step(r)) || length(r) <= 1
 
+function allunique(A::AbstractArray)
+    if length(A) < 2
+        return true
+    elseif length(A) < 32 && A isa StridedArray  # then linear search is certainly faster
+        iter = eachindex(A)
+        I = iterate(iter)
+        while I !== nothing
+            i, s = I
+            a = A[i]
+            for j in Iterators.rest(iter, s)
+                isequal(a, @inbounds A[j]) && return false
+            end
+            I = iterate(iter, s)
+        end
+        return true
+    else
+        invoke(allunique, Tuple{Any}, A)
+    end
+end
+
+function allunique(t::Tuple)
```

Comment:
Why is this specialized for `Tuple` it seem like it would be clearer if this was just a generic fallback.

Reply:
Do you meen Iterators that we know length?
These optimization is for small inputs only, and we aready have a dict based general fallback.

Reply:
It's a case that might be useful, and can be optimised pretty highly. If I widen the StridedArray method to accept tuples, it's a bit slower, like 20ns instead of 6ns at length 10.

For arbitrary iterators, the current hashing method iterated exactly once. We could have a linear search method (at least for iterators of known length) but this will iterate parts of them many times, and I'm not certain that's always a good thing.

Reply:
It's possible the generic iterator method should call `collect` & then the array method, in short enough cases. This allocates but so does the dictionary.